src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h

```diff
@@ -26,6 +27,75 @@
 namespace cldnn {
 namespace onednn {
 
+static std::mutex cacheAccessMutex;
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+std::wstring multiByteCharToWString(const char* str) {
+#ifdef _WIN32
+    int strSize = static_cast<int>(std::strlen(str));
+    int size_needed = MultiByteToWideChar(CP_UTF8, 0, str, strSize, NULL, 0);
+    std::wstring wstrTo(size_needed, 0);
+    MultiByteToWideChar(CP_UTF8, 0, str, strSize, &wstrTo[0], size_needed);
+    return wstrTo;
+#else
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> wstring_encoder;
+    std::wstring result = wstring_encoder.from_bytes(str);
+    return result;
+#endif  // _WIN32
+}
+#endif  // defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+
+
+static std::vector<uint8_t> load_cache_blob_from_disk(std::string path) {
+    std::lock_guard<std::mutex> lock(cacheAccessMutex);
+
+    #if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    std::wstring widefilename = multiByteCharToWString(path.c_str());
+    const wchar_t* filename = widefilename.c_str();
+    FILE *fp = _wfopen(filename, L"rb");
+#else
+    const char* filename = path.c_str();
+    FILE *fp = fopen(filename, "rb");
+#endif
+
+    if (fp) {
+        fseek(fp, 0, SEEK_END);
+        auto sz = ftell(fp);
+        if (sz < 0) {
+            fclose(fp);
+            return {};
+        }
+        auto nsize = static_cast<size_t>(sz);
+
+        fseek(fp, 0, SEEK_SET);
+
+        std::vector<uint8_t> ret(nsize);
+
+        auto res = fread(ret.data(), sizeof(uint8_t), nsize, fp);
+        (void)res;
+        fclose(fp);
+        return ret;
+    }
+
+    return {};
+}
+
+static void store_cache_blob_on_disk(std::string path, std::vector<uint8_t> cache_blob) {
+    std::lock_guard<std::mutex> lock(cacheAccessMutex);
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    std::wstring widefilename = multiByteCharToWString(path.c_str());
+    const wchar_t* filename = widefilename.c_str();
+#else
+    const char* filename = path.c_str();
+#endif
+    std::ofstream out_file(filename, std::ios::out | std::ios::binary);
+    if (out_file.is_open()) {
+        out_file.write(reinterpret_cast<const char*>(&cache_blob[0]), cache_blob.size());
+    } else {
+        throw std::runtime_error("Could not store cl_cache to " + path);
+    }
+}
```

Comment:
Did you review reusing the code from kernels_cache.cpp?

Reply:
Load/Store api are copied from kernels_cache.cpp.
I added it for check the case that could not store cache.